API.md

```diff
@@ -0,0 +1,518 @@
+# Sara Alert API
+
+Sara Alert provides a RESTful API to interact with the system to perform various actions. Actions include reading, writing, and updating monitoree data as well as reading monitoree lab results and monitoree daily reports. The data accepted and returned by the API corresponds to FHIR R4; the FHIR Patient resource is used for monitorees, the Observation FHIR resource is used for monitoree lab results, and the FHIR QuestionaireResponse FHIR resource is used for monitoree daily reports.
+
+- [CapabilityStatement](#cap)
+  - [GET [base]/fhir/r4/CapabilityStatement](#cap-get)
+- [Authenticating](#auth)
+  - [POST [base]/oauth/token](#auth-post)
+- [Reading](#read)
+  - [GET [base]/Patient/[:id]](#read-get-pat)
+  - [GET [base]/Observation/[:id]](#read-get-obs)
+  - [GET [base]/QuestionaireResponse/[:id]](#read-get-que)
+  - [GET [base]/Patient/[:id]/$everything](#read-get-all)
```

Comment:
Couple things wrong with this `$everything` operation.

First, the response I got (using the example `Patient/55`) was invalid:

```json
    {
      "fullUrl": "http://localhost:3000/fhir/r4/Assessment/123",
      "resource": {
        "id": 123,
        "meta": {
          "lastUpdated": "2020-05-11T17:35:46+00:00"
        },
        "status": "completed",
        "subject": {
          "reference": "Patient/55"
        },
        "item": [
          {
            "text": "cough",
            "answer": [
              {
                "valueBoolean": false
              }
            ]
          },
          {
         // truncated
 "resourceType": "QuestionnaireResponse"
      }
```

First, the above resource in the response bundle, is a `QuestionnaireResponse` but the `fullUrl` says that it is an `Assessment`. Bad.

Second, the `QuestionnaireResponse` is not valid, because each `item` is missing a required `linkId` field (cardinality `1..1`).

For example, the `fhir_model` `Resource#validate` method returns:
```
{"entry"=>
  [{"resource"=>
     [{"item"=>
        [{"linkId"=>["Item.linkId: invalid cardinality. Found 0 expected 1..1"]},
         {"linkId"=>["Item.linkId: invalid cardinality. Found 0 expected 1..1"]},
         {"linkId"=>["Item.linkId: invalid cardinality. Found 0 expected 1..1"]},
         {"linkId"=>["Item.linkId: invalid cardinality. Found 0 expected 1..1"]},

.... truncated....
```

Reply:
Good catch, both fixed, and test added.